homework/src/index.js

```diff
@@ -30,18 +45,78 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
-    });
+  function renderRepositoryInfo(info) {
+    const li = createAndAppend('li', ul);
+    createAndAppend('span', li, { id: 'title', text: 'Repository' });
+    const p1 = createAndAppend('p', li);
+    createAndAppend('a', p1, { href: info.RepositoryUrl, target: '_blank', text: info.Repository });
+    const li2 = createAndAppend('li', ul);
+    createAndAppend('span', li2, { id: 'title', text: 'Description:  ' });
+    createAndAppend('p', li2, { text: info.Description });
+    const li3 = createAndAppend('li', ul);
+    createAndAppend('span', li3, { id: 'title', text: 'Forks: ' });
+    createAndAppend('p', li3, { text: info.Forks });
+    const li4 = createAndAppend('li', ul);
+    createAndAppend('span', li4, { id: 'title', text: 'Updated: ' });
+    createAndAppend('p', li4, { text: info.Updated });
+  }
+
+  async function renderContributorsInfo(info) {
+    try {
+      const contributors = await fetchJSON(info.ContributorsUrl);
+      ul2.innerText = '';
+      contributors.forEach(contributor => {
+        const li = createAndAppend('li', ul2);
+        createAndAppend('img', li, { src: contributor.avatar_url });
+        createAndAppend('a', li, { href: contributor.html_url, target: '_blank', text: contributor.login });
+        createAndAppend('span', li, { text: contributor.contributions });
+      });
+    }
+    catch (err) {
+      createAndAppend('div', contributionsInfo, { text: err.message, class: 'alert-error' });
+    }
+  }
+
+  async function main(url) {
+    try {
+      const repositories = await fetchJSON(url);
+      const select = createAndAppend('select', header);
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      const sortedQueries = [];
+      repositories.forEach(repository => {
+        const item = { Repository: repository.name, Description: repository.description, Forks: repository.forks, Updated: repository.updated_at, ContributorsUrl: repository.contributors_url, RepositoryUrl: repository.html_url };
+        sortedQueries.push(item);
+      });
+
+      sortedQueries.forEach((query, index) => {
+        createAndAppend('option', select, { value: index, text: query.Repository });
+      });
+
+      renderRepositoryInfo(sortedQueries[0]);
+
+      renderContributorsInfo(sortedQueries[0]);
```

Comment:
The `renderContributorsInfo()` function really only requires the `contributors_url` property. It is best to give a function just the information that it needs and nothing more. This help to make your code more robust and less error prone. So you could call your function like this:

```js
renderContributorsInfo(repositories[0].contributors_url);
```